riscv-executor/src/lib.rs

```diff
@@ -44,16 +46,13 @@ mod pil;
 
 use crate::profiler::Profiler;
 
-// TODO: we don't do anything with these yet. Idea is to keep info that is to be given to witgen
-#[allow(unused)]
-struct MainOp<F: FieldElement>(&'static str, u32, Vec<F>);
-
 #[derive(Debug)]
 struct SubmachineOp<F: FieldElement> {
     // pil identity id of the link
     identity_id: u64,
-    // these are the RHS values of the lookup (i.e., inside brackets in the PIL lookup)
-    lookup_args: Vec<F>,
+    // these are the RHS values of the lookup (i.e., inside brackets in the PIL lookup).
+    // This is a fixed size to avoid allocations in the common case.
+    lookup_args: [F; 4],
```

Comment:
does this mean we always have a max of 4 elements?

Reply:
for now yes, no calls with more than 4 elements, can be changed later if there's need